src/header/common/content_disposition.rs

```diff
@@ -0,0 +1,384 @@
+                    let mut use_simple_format: bool = false;
+                    if opt_lang.is_none() {
+                        if let Charset::Ext(ref ext) = *charset {
+                            if ext.to_ascii_lowercase() == "utf-8" {
+                                use_simple_format = true;
+                            }
+                        }
+                    }
+                    if use_simple_format {
+                        try!(write!(f, "; filename=\"{}\"",
+                                    match String::from_utf8(bytes.clone()) {
+                                        Ok(s) => s,
+                                        Err(_) => return Err(fmt::Error),
+                                    }));
+                    } else {
+                        let langstr = match *opt_lang {
+                            Some(ref lang) => format!("{}",lang),
+                            None => "".to_owned(),
+                        };
+                        try!(write!(f, "; filename*={}'{}'", charset, langstr));
+                        try!(f.write_str(&*try!(bytes_to_value_chars(bytes))))
+                    }
+                },
+                &DispositionParam::Ext(ref k, ref v) => try!(write!(f, "; {}=\"{}\"", k, v)),
+            }
+        }
+        Ok(())
+    }
+}
+
+fn unquote_value(val: &str) -> String {
+    if val.chars().next() == Some('"') && val.chars().rev().next() == Some('"') {
+        // Unwrap the quotation marks.
+        (&val[1..val.len() - 1]).to_owned()
+    } else {
+        val.to_owned()
+    }
+}
+
+/// Parsing of `ext-value`
+/// https://tools.ietf.org/html/rfc5987#section-3.2
+///
+/// # ABNF
+/// ```plain
+/// ext-value     = charset  "'" [ language ] "'" value-chars
+///               ; like RFC 2231's <extended-initial-value>
+///               ; (see [RFC2231], Section 7)
+///
+/// charset       = "UTF-8" / "ISO-8859-1" / mime-charset
+///
+/// mime-charset  = 1*mime-charsetc
+/// mime-charsetc = ALPHA / DIGIT
+///               / "!" / "#" / "$" / "%" / "&"
+///               / "+" / "-" / "^" / "_" / "`"
+///               / "{" / "}" / "~"
+///               ; as <mime-charset> in Section 2.3 of [RFC2978]
+///               ; except that the single quote is not included
+///               ; SHOULD be registered in the IANA charset registry
+///
+/// language      = <Language-Tag, defined in [RFC5646], Section 2.1>
+///
+/// value-chars   = *( pct-encoded / attr-char )
+///
+/// pct-encoded   = "%" HEXDIG HEXDIG
+///               ; see [RFC3986], Section 2.1
+///
+/// attr-char     = ALPHA / DIGIT
+///               / "!" / "#" / "$" / "&" / "+" / "-" / "."
+///               / "^" / "_" / "`" / "|" / "~"
+///               ; token except ( "*" / "'" / "%" )
+/// ```
+fn parse_ext_value(val: &str) -> ::Result<(Charset, Option<LanguageTag>, Vec<u8>)> {
+
+    // Break into three pieces separated by the single-quote character
+    let parts: Vec<&str> = val.split('\'').collect();
```

Comment:
Could the `val.splitn(3, '\'')` iterator be used, instead of allocating a temporary vec?